bin/reth/src/main.rs

```diff
@@ -14,15 +14,23 @@ use reth_node_builder::{
 };
 use reth_node_ethereum::{node::EthereumAddOns, EthereumNode};
 use reth_provider::providers::BlockchainProvider2;
+use reth_tracing::tracing::warn;
 
 /// Parameters for configuring the engine
 #[derive(Debug, Clone, Args, PartialEq, Eq)]
 #[command(next_help_heading = "Engine")]
 pub struct EngineArgs {
-    /// Enable the engine2 experimental features on reth binary
-    #[arg(long = "engine.experimental", default_value = "false")]
+    /// Enable the experimental engine features on reth binary
+    ///
+    /// DEPRECATED: experimental engine is default now, use --engine.legacy to enable the legacy
+    /// functionality
+    #[arg(long = "engine.experimental", default_value = "true")]
     pub experimental: bool,
 
+    /// Enable the legacy engine on reth binary
+    #[arg(long = "engine.legacy", default_value = "false")]
+    pub legacy: bool,
```

Comment:
what does `--experimental false --legacy false` do? 😄

Reply:
experimental is deprecated and does nothing, so it turns on experimental